File: clojure/resistor-color/src/resistor_color.clj
(ns resistor-color)

(def colors
  ["black" "brown" "red" "orange" "yellow" "green" "blue" "violet" "grey" "white"])

(defn color-code
  "Returns the numerical value associated with the given color"
  [color]
  (->> colors
       (keep-indexed #(when (= color %2) %1))
       first))

File: solutions/lua/resistor-color/1/resistor-color.lua
local codes = {
        'black', --
        'brown', --
        'red', --
        'orange', --
        'yellow', --
        'green', --
        'blue', --
        'violet', --
        'grey', --
        'white'
    }

return {

  colors = function() return codes
end,
  color_code = function(color)
      for i, v in ipairs(codes) do
          if v == color then return i - 1 end
      end
      error(string.format("invalid color: %s", color))
  end,

}
